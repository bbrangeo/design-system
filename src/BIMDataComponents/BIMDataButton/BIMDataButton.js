import BIMDataButton from './BIMDataButton.vue';

export default BIMDataButton;