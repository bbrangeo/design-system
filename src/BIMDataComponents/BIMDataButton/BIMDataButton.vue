<template>
  <button
    class="bimdata-btn"
    :style="{ 'min-width': width, height: height }"
    @click="$emit('click', $event)"
  >
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: "BIMDataButton",
  props: {
    width: {
      type: [Number, String],
      default: "150px",
    },
    height: {
      type: [Number, String],
      default: "32px",
    },
  },
};
</script>

<style lang="scss" scoped>
  // import BIMDATA UTILITIES
  @import "../../assets/scss/utilities/_text.scss";
</style>

<style lang="scss">
  // import BIMDATA VARIABLES
  @import "../../assets/scss/_BIMDataVariables.scss";

  // import BIMDATA MIXINS
  @import "../../assets/scss/mixins/_font-size.scss";

// import BIMDATA STYLE COMPONENT
@import "./_BIMDataButton.scss";
</style>
