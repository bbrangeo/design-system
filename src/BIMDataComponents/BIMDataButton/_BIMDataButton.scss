.bimdata-btn {
  padding: 0;
  display: flex;
  align-items: center;
  background-color: transparent;
  border: none;
  color: $color-primary;
  cursor: pointer;
  @include font-size(12px);
  transition: all 0.1s ease;
  &:focus {
    outline: none;
  }
  /* style BIMDATA BTN RADIUS ------------------ */
  &__radius {
    border-radius: 3px;
  }
  /* style BIMDATA BTN FILL -------------------- */
  &__fill {
    justify-content: center;
    /* style BIMDATA BTN DEFAULT FILL ---------- */
    &--default {
      box-shadow: 0 2px 10px 0 rgba($color-black, 0.1);
    }
    /* style BIMDATA BTN PRIMARY FILL ---------- */
    &--primary {
      background-color: $color-primary;
      color: $color-white;
    }
    /* style BIMDATA BTN GREY FILL ------------- */
    &--grey {
      background-color: $color-border;
    }
    /* style BIMDATA BTN GREY FILL ------------- */
    &--red {
      background-color: $color-red;
      color: $color-white;
    }
    /* style BIMDATA BTN FILL BTN ICON --------- */
    &--btn-icon {
      svg {
        margin: 0 6px;
      }
    }
    /* style BIMDATA BTN FILL ICON ------------ */
    &--icon {
    }
  }
  &__outline {
    justify-content: center;
    border: 1px solid;
    &--default {
      border-color: $color-border;
      color: $color-primary;
    }
    &--grey {
      border-color: $color-grey;
      color: $color-grey;
    }
    &--btn-icon {
    }
    &--icon {
    }
  }
  &__ghost {
    justify-content: flex-start;
    svg {
      display: flex;
      align-items: center;
    }
    &--grey {
      color: $color-text;
    }
    &--btn-icon {
    }
    &--icon {
    }
  }
}
