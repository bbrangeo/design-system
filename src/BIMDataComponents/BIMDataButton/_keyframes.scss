@mixin lightenRipple($color, $growthField, $amount) {
  position: relative;
  &:focus:before {
    content: "";
    width: 100%;
    height: 100%;
    position: absolute;
    border-radius: 100%;
    animation: lightenRipple 0.6s;
  }
  &:focus:active:before {
    content: none;
  }
  @keyframes lightenRipple {
    0% {
      border: $growthField + px solid lighten($color, $amount);
      opacity: 0;
    }
    100% {
      border: 1px solid transparent;
    }
  }
}

@mixin darkenRipple($color, $growthField, $amount) {
  position: relative;
  &:focus:before {
    content: "";
    width: 100%;
    height: 100%;
    position: absolute;
    border-radius: 100%;
    animation: darkenRipple 0.6s;
  }
  &:focus:active:before {
    content: none;
  }
  @keyframes darkenRipple {
    0% {
      border: $growthField + px solid darken($color, $amount);
      opacity: 0;
    }
    100% {
      border: 1px solid transparent;
    }
  }
}
