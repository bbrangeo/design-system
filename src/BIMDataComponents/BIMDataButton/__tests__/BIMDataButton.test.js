import { shallowMount } from "@vue/test-utils";
import BIMDataButton from "../BIMDataButton.vue";

describe("BIMDataButton", () => {
  it("should render component and match snapshot", () => {
    const wrapper = shallowMount(BIMDataButton);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a button with a slot and match snapshot", () => {
    const wrapper = shallowMount(BIMDataButton, {
      slots: {
        default: "<span>test</span>",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a button with a custom width and match snapshot", () => {
    const wrapper = shallowMount(BIMDataButton, {
      propsData: {
        width: "50px",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a button with a custom height and match snapshot", () => {
    const wrapper = shallowMount(BIMDataButton, {
      propsData: {
        height: "50px",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should emit a click event", () => {
    const onClick = jest.fn();
    const wrapper = shallowMount(BIMDataButton, {
      listeners: {
        click: onClick,
      },
    });
    wrapper.trigger("click");
    expect(onClick).toHaveBeenCalled();
  });
});
