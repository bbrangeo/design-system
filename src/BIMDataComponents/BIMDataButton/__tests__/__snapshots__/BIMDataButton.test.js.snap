// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BIMDataButton should render a button with a custom height and match snapshot 1`] = `"<button color=\\"default\\" type=\\"button\\" class=\\"bimdata-btn\\" style=\\"min-width: 32px; min-height: 50px;\\"></button>"`;

exports[`BIMDataButton should render a button with a custom width and match snapshot 1`] = `"<button color=\\"default\\" type=\\"button\\" class=\\"bimdata-btn\\" style=\\"min-width: 50px; min-height: 32px;\\"></button>"`;

exports[`BIMDataButton should render a button with a slot and match snapshot 1`] = `"<button color=\\"default\\" type=\\"button\\" class=\\"bimdata-btn\\" style=\\"min-width: 32px; min-height: 32px;\\"><span>test</span></button>"`;

exports[`BIMDataButton should render component and match snapshot 1`] = `"<button color=\\"default\\" type=\\"button\\" class=\\"bimdata-btn\\" style=\\"min-width: 32px; min-height: 32px;\\"></button>"`;
